Refactored runtest.sh. Mostly renaming variables (including different…

zachmu opened the pull request:
…iating between constants and variables), and extracting some functions.

Signed-off-by: Zach Musgrave <zach@liquidata.co>

zachmu committed:
Refactored runtest.sh. Mostly renaming variables (including differentiating between constants and variables), and extracting some functions.

Signed-off-by: Zach Musgrave <zach@liquidata.co>

Hydrocharged commented on testharness/runtest.sh:
```diff
@@ -26,102 +28,128 @@ if [ -d output/ ]; then
+    echo "End:------- $suiteName/$testName" >&2
+
+    echo "Cleaning up logs and reject files"
+    cleanupRejects "$suiteName"
+    cleanupLogs "$suiteName"
```
We'll be running `cleanupRejects` and `cleanupLogs` after every single test, which seems a bit excessive. Perhaps it would be better to have this in `runSuite`, and also further down when you run a single test.

zachmu replied:
Not concerned, it's a miniscule performance hit (same number of file moves, one additional directory listing per test)

Hydrocharged commented on testharness/runtest.sh:
```diff
@@ -26,102 +28,128 @@ if [ -d output/ ]; then
-  FULLSUITENAME="$1"
-  ARG1="${FULLSUITENAME%/*}"
-  ARG2="${FULLSUITENAME#*/}"
-fi
```
I like the change that allows for running entire suites, but what about the single-test case where it's passed as "SUITE/TEST" instead of "SUITE TEST"? It's a bit more significant since the output formats tests in that way, so preserving that format for input allows one to just copy and paste it in. Sure it's a small modification, but I think it's worth including.

zachmu replied:
Sure I'll make that change

Hydrocharged reviewed (commented):
I'm not sure I understand the decision behind making some variables camelCase and leaving others in SNAKE_CASE? I thought that maybe all variables local to the script were camelCase, but that's not true for some such as `FILESDIR`.

zachmu commented:
The naming changes are to differentiate the constants (UPPERCASE) from the variables (camelCase)

zachmu committed:
Work in progress. Changes to result parsing logic

Signed-off-by: Zach Musgrave <zach@liquidata.co>

zachmu committed:
Added tests for parseoutput.go

Signed-off-by: Zach Musgrave <zach@liquidata.co>

zachmu committed:
Added test files for parseoutput.go

Signed-off-by: Zach Musgrave <zach@liquidata.co>

zachmu committed:
PR feedback: make it possible to still specify input as suiteName/testName

Signed-off-by: Zach Musgrave <zach@liquidata.co>

zachmu committed:
Added an error message if the test named doesn't exist, remove debug mysqltest command

Signed-off-by: Zach Musgrave <zach@liquidata.co>

zachmu commented:
PTAL when you have a minute. The big changes from last time:

1) Changed how the timing output and parsing is handled
2) Added a unit test for the parser
3) Addressed PR feedback.

Hydrocharged commented on testharness/parseoutput.go:
```diff
@@ -72,96 +86,118 @@ func main() {
+				testName := detailLine[len(startMarker):]
+				var ok bool
+				if state.TestResult, ok = results[testName]; !ok {
+					log.Fatalf("Uncrecognized test name %v on line %v", testName, detailLine)
```
```suggestion
					log.Fatalf("Unrecognized test name %v on line %v", testName, detailLine)
```

Hydrocharged commented on testharness/parseoutput.go:
```diff
@@ -72,96 +86,118 @@ func main() {
-				state.TestResult.Reject, err = ReadFile(rejectFileName)
+			state.TestResult.ElapsedSeconds = int64(state.LastTimestamp.Sub(state.FirstTimestamp).Seconds())
+
+
```
So much whitespace 😭

zachmu committed:
Update testharness/parseoutput.go

Co-Authored-By: Daylon Wilkins <Hydrocharged@users.noreply.github.com>
